docs/DeveloperGuide.md

```diff
@@ -500,85 +500,96 @@ The other commands `stats curr` and `stats housing` have a similar execution pat
 
 </div>
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
 
-![UndoRedoState3](images/UndoRedoState3.png)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index 0, pointing to the initial AddressBook state, then there are no previous AddressBook states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather
-than attempting to perform the undo.
+#### Design considerations:
 
-</div>
+**Aspect: How undo executes:**
 
-The following sequence diagram shows how the undo operation works:
+* **Alternative 1 (current choice):** Saves a backup of the entire address book.
+  * Pros: Easy to implement and not too memory intensive.
+  * Cons: Can only save one previous instance of the address book.
 
-![UndoSequenceDiagram](images/UndoSequenceDiagram.png)
+* **Alternative 2:** Individual commands knows how to undo by itself.
+  * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
+  * Cons: We must ensure that the implementation of each individual command are correct.
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+* **Alternative 3:** Saves multiple backups of the entire address book.
+  * Pros: Easy to implement.
+  * Cons: Memory intensive and may have performance issues in terms of memory usage.
 
-</div>
+### Reset feature
 
-The `redo` command does the opposite — it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+#### Implementation
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+The undo mechanism allows users to revert the last executed command in the address book. This feature is facilitated by the `UndoCommand`, `AddressBookParser`, and `Model` classes. The undo feature is implemented using the following components and operations:
```

Comment:
u might want to update this line, seems to be that of the undo feature

Reply:
Oops. Alrighty.